File: physics.py
import math
class physics():
	def __init__(self, distance, velocity, acceleration, time) -> None:
		self.distance = distance
		self.velocity = velocity
		self.acceleration = acceleration
		self.time = time
		
	def get_velocity(self, distance, velocity, acceleration, time) -> float:
		if self.acceleration == 0:
			return (self.distance/self.time)+self.velocity
		else:
			return (self.acceleration*self.time)+self.velocity
	def get_acceleration(self, distance, velocity, acceleration, time) -> float:
		if self.velocity == 0:
			return (self.distance/math.pow(self.time, 2))+self.acceleration
		else:
			return self.velocity/self.time
	def get_distance(self, distance, velocity, acceleration, time) -> float:
		return (self.velocity*self.time)+(((self.acceleration)*(self.time))/2)
	def get_Rotational_Velocity(self, distance, velocity, acceleration, time, radius):
		if self.acceleration == 0:
			return self.velocity*radius
		elif self.velocity == 0:
			return (radius*self.acceleration*self.time)
		else:
			return (self.distance*radius)/self.time
	def get_Total_Energy(self, distance, velocity, acceleration, time, height, mass) -> list:
		if self.acceleration == 0 and self.velocity == 0:
			kineticEnergy = mass*(0.5)*math.pow((self.distance/self.time), 2)
		elif self.velocity != 0 and self.acceleration == 0:
			kineticEnergy = (0.5)*math.pow(self.velocity, 2)*mass
		else:
			kineticEnergy = mass*0.5*math.pow(self.acceleration*self.time, 2)
		potentialEnergy = mass*height*9.81
		totalEnergy = potentialEnergy+kineticEnergy
		return [potentialEnergy, kineticEnergy, totalEnergy]
	

File: Physics.cpp
#include <tgmath.h>
#include <iostream>
#include <vector>
#include <stdio.h>

using namespace std;

std::vector<float *> init(){
	float ptime;
	float displacement;
	float velocity;
	float acceleration;
	std::cout << "Enter the change in time: ";
	std::cin >> ptime;
	std::cout << "\n";
	std::cout << "Enter the change in displacement: ";
	std::cin >> displacement;
	std::cout << "\n";
	std::cout << "Enter the velocity or its change: ";
	std::cin >> velocity;
	std::cout << "\n";
	std::cout << "Enter the acceleration or its change: ";
	std::cin >> acceleration;
	std::cout << "\n";
	float* ptrTime = &ptime;
	float* ptrDisp = &displacement;
	float* ptrVel = &velocity;
	float* ptrAcc = &acceleration;
	std::vector<float *> pointerArray = {ptrTime, ptrDisp, ptrVel, ptrAcc};
	return pointerArray;
}
	float get_time(float time, float displacement, float velocity, float acceleration){
		if (acceleration != 0 && displacement != 0){
			return sqrt(((2*displacement)/acceleration));
		}else if (velocity != 0 && displacement != 0){
			return displacement/velocity;
		}else{
			return time;
		}
	}
	float get_displacement(float time, float displacement, float velocity, float acceleration){
		if (acceleration != 0 || velocity !=0){
			return (velocity*time+(0.5*(acceleration*time*time)));
		}else{
			return displacement;
		}
	}
	float get_velocity(float time, float displacement, float velocity, float acceleration){
		if (time != 0){
			return (displacement/time)+(acceleration*time);
		}else{
			return velocity;
		}
	}
	float get_acceleration(float time, float displacement, float velocity, float acceleration){
		if (velocity != 0 && time != 0){
			return velocity/time;
		}else if (displacement != 0 && time != 0){
			return displacement/(time*time);
		}else{
			return acceleration;
		}
	}

int main(){
	std::vector<float *> pointerArray = init();
	float* ptrTime = pointerArray[0];
	float* ptrDisplacement = pointerArray[1];
	float* ptrVelocity = pointerArray[2];
	float* ptrAcceleration = pointerArray[3];
	string wtfDoYouWant;
	std::cout << "Are you calculating for acceleration, velocity, displacement, or time?\n";
	std::cin >> wtfDoYouWant;
	if (wtfDoYouWant == "time"){
		float time = get_time(*ptrTime, *ptrDisplacement, *ptrVelocity, *ptrAcceleration);
		std::cout << time;
	}else if (wtfDoYouWant == "displacement"){
		float displacement = get_displacement(*ptrTime, *ptrDisplacement, *ptrVelocity, *ptrAcceleration);
		
	}else if (wtfDoYouWant == "velocity"){
		float velocity = get_velocity(*ptrTime, *ptrDisplacement, *ptrVelocity, *ptrAcceleration);
		
	}else if (wtfDoYouWant == "acceleration"){
		float acceleration = get_acceleration(*ptrTime, *ptrDisplacement, *ptrVelocity, *ptrAcceleration);
		
	}
}
